### PR TITLE
Remove Player from serialized milestomes and awards.

### PR DESCRIPTION
**Pull request by kberg**
Remove Player from serialized milestomes and awards.

This reduces the example game I have been using from its original size
of 1.950M bytes to 340K to 27.5K

This is database storage, mind you.

### CODE REVIEW

**Comment by kberg** [comment]
This ends my serialization work.

File: src/awards/FundedAward.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+export function deserializeFundedAwards(
+  fundedAwards: Array<SerializedFundedAward>,
+  players: Array<Player>,
+  awards: Array<IAward>): Array<FundedAward> {
+  return fundedAwards.map((element: SerializedFundedAward) => {
+    const awardName = element.award?.name !== undefined ? element.award.name : element.name;
+    if (awardName === undefined) {
+      throw new Error('Milestone name not found');
```
Comment (nwai90) [inline src/awards/FundedAward.ts:33]: This should probably be `Award name not found`

File: src/awards/FundedAward.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    const award: IAward | undefined = awards.find((award) => award.name === awardName);
+    if (award === undefined) {
+      throw new Error(`Award ${awardName} not found when rebuilding Claimed Milestone`);
```
Comment (nwai90) [inline src/awards/FundedAward.ts:37]: Similarly for these, should they be `Award ${awardName} not found when rebuilding Funded Award`?

Comment (kberg) [reply src/awards/FundedAward.ts:37]: Ah, yes, thanks.

File: src/awards/FundedAward.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import {Player, PlayerId} from '../Player';
+
+export interface FundedAward {
+  award: IAward;
+  player: Player;
+}
+
+// TODO(kberg): remove references to award and player by 2020-02-01
+export interface SerializedFundedAward {
+  name?: string;
```
Comment (bafolts) [inline src/awards/FundedAward.ts:11]: With all the enums we have in place we don't have `AwardName`. Good to know.

File: src/awards/FundedAward.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    if (playerId === undefined) {
+      throw new Error(`Player ID not found when rebuilding Funded Award ${awardName}`);
+    }
+    const player = players.find((player) => player.id === playerId);
+    if (player === undefined) {
+      throw new Error(`Player ${playerId} not found when rebuilding Funded Award ${awardName}`);
+    }
+
+    return {
+      award: award,
```
Comment (bafolts) [inline src/awards/FundedAward.ts:50]: Fun fact, typescript will allow you to do `return { award, player }` here.

Comment (kberg) [reply src/awards/FundedAward.ts]: OK made the change.

**Review by bafolts — APPROVED** [review]
Looks good.

**Pull request merged**